tests/network/link_state.go

```diff
@@ -0,0 +1,226 @@
+/*
+ * This file is part of the kubevirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2025 Red Hat, Inc.
+ *
+ */
+
+package network
+
+import (
+	"context"
+	"fmt"
+
+	"kubevirt.io/kubevirt/pkg/apimachinery/patch"
+
+	"k8s.io/apimachinery/pkg/types"
+
+	"kubevirt.io/kubevirt/pkg/pointer"
+
+	"time"
+
+	"kubevirt.io/kubevirt/tests/framework/kubevirt"
+
+	"kubevirt.io/kubevirt/tests/libnet"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	libvmici "kubevirt.io/kubevirt/pkg/libvmi/cloudinit"
+	"kubevirt.io/kubevirt/tests/libnet/cloudinit"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	v1 "kubevirt.io/api/core/v1"
+
+	"kubevirt.io/kubevirt/pkg/libvmi"
+	. "kubevirt.io/kubevirt/tests/framework/matcher"
+
+	"kubevirt.io/kubevirt/tests/console"
+	"kubevirt.io/kubevirt/tests/libvmifact"
+	"kubevirt.io/kubevirt/tests/testsuite"
+)
+
+const (
+	bridgeInterfaceName2 = "bridge2"
+	bridgeInterfaceName3 = "bridge3"
+	mac1                 = "02:4d:8b:00:00:3a"
+	mac2                 = "02:4d:8b:00:00:4b"
+	mac3                 = "02:4d:8b:00:00:5c"
+	regexLinkStateUP     = "'state[[:space:]]+UP'"
+	regexLinkStateDOWN   = "'state[[:space:]]+DOWN'"
+	ipLinkTemplate       = "ip -one link | grep %s | grep -E %s\n"
+)
+
+var _ = SIGDescribe("interface state up/down", Ordered, func() {
+
+	var vmName types.NamespacedName
+	var netDefault *v1.Network
+	var net2 *v1.Network
+	var net3 *v1.Network
+	var ifaceDefault *v1.Interface
+	var iface2 *v1.Interface
+	var iface3 *v1.Interface
+
+	Context("VM with one link up, one link down, one hotplugged down", func() {
```

Comment:
Please consider breaking the one big test into few smaller ones.
We could optimize for execution speed after having a correct and stable test.